nixos/modules/services/networking/seafile.nix

```diff
@@ -4,33 +4,50 @@ let
   cfg = config.services.seafile;
   settingsFormat = pkgs.formats.ini { };
 
-  ccnetConf = settingsFormat.generate "ccnet.conf" cfg.ccnetSettings;
+  ccnetConf = settingsFormat.generate "ccnet.conf" (cfg.ccnetSettings // {
+    Database = {
+      ENGINE = "mysql";
+      UNIX_SOCKET = "/var/run/mysqld/mysqld.sock";
+      DB = "ccnet_db";
+      CONNECTION_CHARSET = "utf8";
+    };
+  });
```

Comment:
wouldn't this be better as a default for the `ccnetSettings` submodule? having this basically forced internally could be pretty annoying if one wanted to run this with a remote database for example. maybe these options could be made explicit to better handle service settings for cases like that as well

Reply:
It should be done in a way, that the settings are merged properly and not overwritten if one value is changed.

Reply:
I think this would be better as a second PR, as this one is getting pretty big already. Handling different database settings in the 2 formats and then also in the migration script would take non-insignificant effort/time, I think.